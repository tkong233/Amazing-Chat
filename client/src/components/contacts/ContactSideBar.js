import React, { Component, useEffect } from "react";
import { connect } from "react-redux";

import SuggestionList from './SuggestionList';
import ContactList from './ContactList';
import SearchBar from './SearchBar';
import { getSuggestion, getContacts, getAllUsers } from '../../actions/contactActions';
import { connectSocket } from '../../actions/chatActions';

import './Contact.css';

const ENDPOINT = 'localhost:5000/';

const ContactSideBar = (props) => {
  useEffect(() => {
    props.getSuggestion(props.user.email);
    props.getContacts(props.user.email);
    props.getAllUsers();
  }, [])

  return (
    <div
      className='contact-side-bar'
      data-test = "DashboardComponent">
      <SearchBar/>
      <SuggestionList/>
      <ContactList/>
    </div>
  );
}

const mapStateToProps = state => ({
  user: state.auth.user,
  contact: state.contact,
});


export default connect(
  mapStateToProps, 
  {
    getSuggestion,
    getContacts,
    getAllUsers,
    connectSocket,
  }
)(ContactSideBar);