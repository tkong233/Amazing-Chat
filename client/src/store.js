import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { persistStore, persistReducer } from 'redux-persist'
import rootReducer from "./reducers";
import storage from 'redux-persist/lib/storage' // defaults to localStorage for web

const initialState = {};

const persistConfig = {
  key: 'root',
  storage,
}

const middleware = [thunk];

const persistedReducer = persistReducer(persistConfig, rootReducer, initialState);

const store = createStore(

  persistedReducer,
  compose(
    applyMiddleware(...middleware),
    (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__()) ||
      compose
  )
);

const persistor = persistStore(store);

export {
  store,
  persistor,
  middleware
};



