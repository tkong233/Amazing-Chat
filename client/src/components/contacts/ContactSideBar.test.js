import React from 'react';
import ContactSideBar from './ContactSideBar';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16'; 
import { applyMiddleware, createStore } from 'redux';
import {middleware} from '../../store';
import rootReducer from '../../reducers/index'

Enzyme.configure({
    adapter: new EnzymeAdapter(),
});

const setUp = (initialState={}) =>{
    const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
    const store = createStoreWithMiddleware(rootReducer, initialState);
    const wrapper = shallow(<ContactSideBar store={store}/>).dive();
    // console.log(wrapper.debug());
    return wrapper;
}

describe('ContactSideBar Component', ()=>{
    describe('Checking PropTypes', ()=>{
        it('Should not throw a warning', ()=>{
            const expectedProps = {
                contact: {
                    users:{},
                    contacts:{}
                },
                auth: {
                    user:{}
                },
            }
            // eslint-disable-next-line react/forbid-foreign-prop-types
            const propsErr = checkPropTypes(ContactSideBar.propTypes, expectedProps, 'props', ContactSideBar.name);
            expect(propsErr).toBeUndefined();
        })
    });
    describe('Renders', ()=>{
        let wrapper;
        beforeEach(()=>{
            const initialState = {
                auth:{user: {
                    name: 'testuser',
                    email: 'testuser@test.com',
                    profile_picture: 'testpicture'
                }}
            };
            wrapper = setUp(initialState);
        });

        it('Should render without errors', ()=>{
            const container = wrapper.find(`[data-test='ContactSideBarComponent']`);
            expect(container.length).toBe(1);
        });

        it('Snapshot testing', ()=>{
            expect(wrapper).toMatchSnapshot();
        });


    });
});