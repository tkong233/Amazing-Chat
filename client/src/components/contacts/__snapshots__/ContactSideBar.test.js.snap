// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactSideBar Component Renders Snapshot testing 1`] = `ShallowWrapper {}`;
